Release 0.35.0

tkoyama010 opened the pull request:
:rocket: Let's publish the next minor release of PyVista! :rocket:

As it is said in #2882 . This release is needed as support of Python3.10 ASAP.

Thanks again to all of those who tirelessly contribute to this project!

---
## Waiting on the following Issues or PRs:

@pyvista/developers Any other Issues or PR you want to add to this release?

- [ ] #2901
- [ ] #2896
- [ ] #2888
- [ ] #2416

adeak commented:
I've added https://github.com/pyvista/pyvista/pull/2897 (reviewed and will be merged soon though) and https://github.com/pyvista/pyvista/pull/2908 (minor but causes an ugly figure in the docs, easy to review).

akaszynski commented on pyvista/_version.py:
```diff
@@ -15,7 +15,7 @@
 """
 # major, minor, patch
-version_info = 0, 35, 'dev0'
+version_info = 0, 35, 0
```
What we should do is bump to the next dev version in the source and then bump to `0.35.0` in the release branch.

```suggestion
version_info = 0, 36, 'dev0'
```

tkoyama010 replied:
Thanks for this. I reflected on this in #2911 .

MatthewFlamm commented:
You can hold off on the `DataSet.cell_point_ids` #2897 and `DataSet.(is_point_inside_cell)` #2905.  These are thematically similar and the latter is not yet ready to go.

tkoyama010 commented:
> Plan for 0.35.0 to be out for [SciPy 2022](https://www.scipy2022.scipy.org/) since we have a lot of new features, but the Python 3.10 issue should be fixed immediately.

Great idea! I added to the contents.

> You can hold off on the DataSet.cell_point_ids https://github.com/pyvista/pyvista/pull/2897 and DataSet.(is_point_inside_cell) https://github.com/pyvista/pyvista/pull/2905. These are thematically similar and the latter is not yet ready to go.

Great! I added it to the list. Really looking forward to this feature.

MatthewFlamm commented:
I am now on vacation, so unless I can somehow get codespaces to work for a dev pyvista environment, I won't be able to work on #2905 until around July 10.  So this one can be left out of 0.35.0 if a release is planned sooner.

akaszynski commented:
> I am now on vacation, so unless I can somehow get codespaces to work for a dev pyvista environment, I won't be able to work on #2905 until around July 10. So this one can be left out of 0.35.0 if a release is planned sooner.

Not that I'm encouraging doing "work" on vacation, but if you want to, https://gitpod.io/ is the way to go. It's even free up to 50 and works great. First time using it was at PyCon, and there was even someone else there who used it because their laptop died enroute. They were able to get the full development environment setup, all for free!

MatthewFlamm commented:
I find working on open source projects relaxing somehow.

> but if you want to, https://gitpod.io/ is the way to go. 

Thanks for the suggestion.  I believe codespaces is also free for public repos too?  I am a newbie at working with docker and so on with these tools.  I'm getting errors like `ImportError: libGL.so.1 ...` in either tool.  So I know I've got to install things into the container.  This might be a good topic for the developer docs.

akaszynski reviewed (approved):
Thanks everyone for your hard work!
